src/Projectile.java

```diff
@@ -4,46 +4,26 @@
 public class Projectile extends DefaultEntity {
 
     protected int lifetime;
-
-    protected int speed;
-
     protected int age;
 
-    public Projectile(Vector position, Vector velocity, int speed, int lifetime) {
+    public Projectile(Presence presence, int lifetime) {
         // Call super constructor
-        super(
-            new Vector(position),
-            new Vector(
-                    velocity.x() * speed,
-                    velocity.y() * speed
-                )
-            );
-
-        this.speed = speed;
-
+        super(presence);
+        // Reference lifetime
         this.lifetime = lifetime;
 
         // Initalizie age
         this.age = this.lifetime;
     }
 
-    public Projectile(Vector position, Vector velocity) {
-        this(position, velocity, 5, 10);
-    }
-
-
-    public Projectile(Vector position, Vector velocity, int speed) {
-        this(position, velocity, speed, 10);
-    }
-
     @Override
     public String getName() {
         return "PROJECTILE (" + this.age + ")";
     }
 
     @Override
     public void update(String command) {
-        this.age -= this.speed;
+        this.age -= 1;
```

Comment:
@HN67 I guess for here, we just need to convert the velocity vector from presence into a scalar for speed? Or do you think there might be a better way to do it?

Reply:
Sorry, which constant do you mean? Also, could you expand on your goal for the speed rework of Projectile you did?

Reply:
I did that speed rework before I saw the `presence` branch. So, you can disregard that. The constant that I'm talking about is the integer 1 which subtracts from the age of the projectile on every update. I was just wondering if we make this more dynamic. Because the the age is going to deprecate depending on the velocity of the Projectile, right?

Reply:
Also, as you can see, I force pushed last night to get right of the changes that I made when I wasn't aware of the presence branch.

Reply:
I don't know if we even need to make the age depreciate based on the velocity, but we could do the same thing that we were doing with passing the speed as an int, and then scaling the presence's velocity by that speed. Definitely something that would be easy to implement later. If you really think you need to force push go ahead, but I think resolving the conflicts is probably better.

Reply:
Yeah, had to do both actually. Remove my previous commits and resolve conflicts.